internal/go_repository.bzl

```diff
@@ -387,8 +388,16 @@ def _generate_package_info(*, importpath, version):
     package_name = importpath
 
     # TODO: Consider adding support for custom remotes.
-    package_url = "https://" + importpath if version else None
+    package_url = "https://{}".format(importpath) if version else None
     package_version = version.removeprefix("v") if version else None
+    # See specification:
+    # https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#golang
+    # scheme:type/namespace/name@version?qualifiers#subpath
+    purl = "pkg:golang/{namespace_and_name}{version}".format(
+        namespace_and_name = importpath,
+        version = "@{}".format(version) if version else "",
```

Comment:
The purl spec doesn't contain any information or examples on how version numbers are encoded. Do you know of any consumers that support this format?

Reply:
Based on https://github.com/package-url/purl-spec/issues/63

- ORT, Syft, and SCTK seem to be some consumers that read it
- There are some thorny cases
- version as `@1.2.3` does not seem contentious

Reply:
Wish I hadn't started reading that thread, why can't we have nice (and simple) things for once?

But I agree that the current form is definitely the most pleasant one, so let's go with it.